lib/rest-adapter.js

```diff
@@ -143,17 +143,45 @@ RestAdapter.prototype.invoke = function(method, ctorArgs, args, callback) {
   });
 };
 
-RestAdapter.prototype.getRestMethodByName = function(name) {
+/**
+ * creates the rest method by name cache map.
+ * @returns {Object} map of rest method name to rest method
+ */
+RestAdapter.prototype._createRestMethodByNameCache = function() {
+  var restMethodByNameMap = {};
   var classes = this.getClasses();
   for (var i = 0; i < classes.length; i++) {
     var restClass = classes[i];
     for (var j = 0; j < restClass.methods.length; j++) {
       var restMethod = restClass.methods[j];
-      if (restMethod.fullName === name) {
-        return restMethod;
-      }
+      restMethodByNameMap[restMethod.fullName] = restMethod;
     }
   }
+  return restMethodByNameMap;
+};
+
+RestAdapter.prototype.getRestMethodByName = function(name) {
+  var cacheCreatedNow = false;
+  if (!this._cachedRestMethodsByName) {
+    this._cachedRestMethodsByName = this._createRestMethodByNameCache();
+    cacheCreatedNow = true;
+  }
+  var ret = this._cachedRestMethodsByName[name];
+  if (!ret && !cacheCreatedNow) {
+    // check if it has been added recently
+    this._cachedRestMethodsByName = this._createRestMethodByNameCache();
+    ret = this._cachedRestMethodsByName[name];
+  }
+  if (ret && !ret.sharedMethod.sharedClass.isMethodEnabled(ret.sharedMethod)) {
+    ret = null;
+  }
+  return ret;
+};
+
+RestAdapter.prototype.invalidateRestMethodCache = function() {
```

Comment:
Since you are already checking for disabled methods on L175 above (where I commented because I was confused), I think there is no need to have `invalidateRestMethodCache` and I am proposing to remove this method.